File: clients/python/examples/vectorize-alter-table-enable.py
# tag::imports-definition[]
from astrapy import DataAPIClient
from astrapy.info import (
    AlterTableAddVectorize,
    VectorServiceOptions,
)
# end::imports-definition[]

# tag::pre-table-definition[]

# Get an existing table
client = DataAPIClient("APPLICATION_TOKEN")
database = client.get_database("API_ENDPOINT")
table = database.get_table("TABLE_NAME")

# end::pre-table-definition[]

# tag::table-definition-external-provider[]
# Configure an embedding provider for a column
table.alter(
    AlterTableAddVectorize(
        columns={
            "VECTOR_COLUMN_NAME": VectorServiceOptions(
                provider="{embedding-provider-name-api}",
                model_name="MODEL_NAME",
                authentication={
                    "providerKey": "API_KEY_NAME",
                },
            ),
        },
    )
)
# end::table-definition-external-provider[]

# tag::table-definition-hugging-face-dedicated[]
# Configure an embedding provider for a column
table.alter(
    AlterTableAddVectorize(
        columns={
        "VECTOR_COLUMN_NAME": VectorServiceOptions(
                provider="{embedding-provider-name-api}",
                model_name="{embedding-provider-model-name-api}",
                authentication={
                    "providerKey": "API_KEY_NAME",
                },
                parameters={
                    "endpointName": "ENDPOINT_NAME",
                    "regionName": "REGION",
                    "cloudName": "CLOUD_PROVIDER",
                },
            ),
        },
    )
)
# end::table-definition-hugging-face-dedicated[]

# tag::table-definition-openai[]
# Configure an embedding provider for a column
table.alter(
    AlterTableAddVectorize(
        columns={
            "VECTOR_COLUMN_NAME": VectorServiceOptions(
                provider="{embedding-provider-name-api}",
                model_name="MODEL_NAME",
                authentication={
                    "providerKey": "API_KEY_NAME",
                },
                parameters={
                    "organizationId": "ORGANIZATION_ID",
                    "projectId": "PROJECT_ID",
                },
            ),
        },
    )
)
# end::table-definition-openai[]

# tag::table-definition-azure-openai[]
# Configure an embedding provider for a column
table.alter(
    AlterTableAddVectorize(
        columns={
            "VECTOR_COLUMN_NAME": VectorServiceOptions(
                provider="{embedding-provider-name-api}",
                model_name="MODEL_NAME",
                authentication={
                    "providerKey": "API_KEY_NAME",
                },
                parameters={
                    "resourceName": "RESOURCE_NAME",
                    "deploymentId": "DEPLOYMENT_ID",
                },
            ),
        },
    )
)
# end::table-definition-azure-openai[]

# tag::table-definition-hosted-provider[]
# Configure an embedding provider for a column
table.alter(
    AlterTableAddVectorize(
        columns={
            "VECTOR_COLUMN_NAME": VectorServiceOptions(
                provider="{embedding-provider-name-api}",
                model_name="{embedding-provider-model-name-api}",
            ),
        },
    )
)
# end::table-definition-hosted-provider[]

File: clients/java/examples/vectorize-alter-table-enable.java
// tag::opening[]
import com.datastax.astra.client.DataAPIClient;
import com.datastax.astra.client.core.vector.SimilarityMetric;
import com.datastax.astra.client.core.vectorize.VectorServiceOptions;
import com.datastax.astra.client.tables.commands.AlterTableAddVectorize;
import com.datastax.astra.client.tables.definition.rows.Row;
import com.datastax.astra.client.tables.Table;

import java.util.HashMap;
import java.util.Map;

public class Example {

  public static void main(String[] args) {
    // Get an existing table
    Table<Row> table =
        new DataAPIClient("APPLICATION_TOKEN")
            .getDatabase("API_ENDPOINT")
            .getTable("TABLE_NAME");

// end::opening[]

// tag::add-external-provider[]
    // Configure an embedding provider for a column
    AlterTableAddVectorize alterOperation = new AlterTableAddVectorize()
        .columns(
            Map.of(
                "VECTOR_COLUMN_NAME",
                new VectorServiceOptions()
                    .provider("{embedding-provider-name-api}")
                    .modelName("MODEL_NAME")
                    .authentication(Map.of("providerKey", "API_KEY_NAME"))
            )
        );
// end::add-external-provider[]

// tag::add-hugging-face-dedicated[]
    // Define parameters for the embedding provider
    Map<String, Object > params = new HashMap<>();
    params.put("endpointName", "ENDPOINT_NAME");
    params.put("regionName", "REGION");
    params.put("cloudName", "CLOUD_PROVIDER");

    // Configure an embedding provider for a column
    AlterTableAddVectorize alterOperation = new AlterTableAddVectorize()
        .columns(
            Map.of(
                "VECTOR_COLUMN_NAME",
                new VectorServiceOptions()
                    .provider("{embedding-provider-name-api}")
                    .modelName("{embedding-provider-model-name-api}")
                    .authentication(Map.of("providerKey", "API_KEY_NAME"))
            )
        );
// end::add-hugging-face-dedicated[]

// tag::add-openai[]
    // Define parameters for the embedding provider
    Map<String, Object > params = new HashMap<>();
    params.put("organizationId", "ORGANIZATION_ID");
    params.put("projectId", "PROJECT_ID");

    // Configure an embedding provider for a column
    AlterTableAddVectorize alterOperation = new AlterTableAddVectorize()
        .columns(
            Map.of(
                "VECTOR_COLUMN_NAME",
                new VectorServiceOptions()
                    .provider("{embedding-provider-name-api}")
                    .modelName("MODEL_NAME")
                    .authentication(Map.of("providerKey", "API_KEY_NAME"))
                    .parameters(params)
            )
        );
// end::add-openai[]

// tag::add-azure-openai[]
    // Define parameters for the embedding provider
    Map<String, Object > params = new HashMap<>();
    params.put("resourceName", "RESOURCE_NAME");
    params.put("deploymentId", "DEPLOYMENT_ID");


    // Configure an embedding provider for a column
    AlterTableAddVectorize alterOperation = new AlterTableAddVectorize()
        .columns(
            Map.of(
                "VECTOR_COLUMN_NAME",
                new VectorServiceOptions()
                    .provider("{embedding-provider-name-api}")
                    .modelName("MODEL_NAME")
                    .authentication(Map.of("providerKey", "API_KEY_NAME"))
                    .parameters(params)
            )
        );
// end::add-azure-openai[]

// tag::add-hosted-provider[]
    // Configure an embedding provider for a column
    AlterTableAddVectorize alterOperation = new AlterTableAddVectorize()
        .columns(
            Map.of(
                "VECTOR_COLUMN_NAME",
                new VectorServiceOptions()
                    .provider("{embedding-provider-name-api}")
                    .modelName("{embedding-provider-model-name-api}")
            )
        );
// end::add-hosted-provider[]

//tag::closing[]

    table.alter(alterOperation);
  }
}
//end::closing[]
